codec-http2/src/test/java/io/netty/handler/codec/http2/Http2FrameCodecTest.java

```diff
@@ -641,6 +647,30 @@ public void operationComplete(ChannelFuture future) throws Exception {
         assertTrue(listenerExecuted.get());
     }
 
+    @Test
+    public void upgradeEventNoRefCntError() throws Http2Exception {
+        frameListener.onHeadersRead(http2HandlerCtx, Http2CodecUtil.HTTP_UPGRADE_STREAM_ID, request, 31, false);
+
+        final FullHttpRequest request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "/");
+        final HttpServerUpgradeHandler.UpgradeEvent upgradeEvent = mock(HttpServerUpgradeHandler.UpgradeEvent.class);
```

Comment:
why do we mock this?  can we construct a real upgrade event here?

Reply:
the constructor is package-private.